support/run-tests.sh

```diff
@@ -40,17 +41,31 @@ done < <(find . -name '*.test' -print0)
 
 additionalQemuArgs=""
 
-supportKVM=$(grep -c -E 'vmx|svm' /proc/cpuinfo || echo "0")
-if [ "$supportKVM" -ne 0 ]; then
+supportKVM=$(grep -c -E 'vmx|svm' /proc/cpuinfo || true)
```

Comment:
This was previously buggy and had printed some error about comparing numbers with strings or something along those lines (don't have the exact error handy anymore).

Reply:
`-ne` is an arithmetic operator and compares integers. So if `$supportKVM` is empty, you compare a string with a value and get an error. The change to `|| true` doesn't really help if netiher `vmx` nor `svm` can be found. The `||` just suppresses an error and script termination if opening `/proc/cpuinfo` fails or if `grep` fails for other reasons.
So below, we should check for a non-empty string.

Reply:
Note the `-c` on the `grep` invocation: it makes `grep` print the number of matches instead. If the result is 0, grep will additionally set an error exit. In that case the previous code would yield the following:

```console
$ supportKVM=$(grep -c -E 'vmx|svm' /proc/cpuinfo || echo "0")
$ echo $supportKVM
0
0
```

Which is why this failed, I think. The second `0` was redundant.

Reply:
The `-c` setting a 1 status is interesting, didn't see that.

But still with the `|| true`, if `/proc/cpuinfo` doesn't exist or isn't readable, `supportKVM` will be an empty string. So you may want to treat this as being "0".

With `|| echo "0"`, `supportKVM` is set to "0" in this case.

Reply:
But `|| echo "0"` will result in two `0`s being printed in case `/proc/cpuinfo` does exist -- see my code block in the previous comment. I'll just throw out the `-c` and check for empty string.

Reply:
Sorry, I either failed to reproduce here or my eyes filtered out the second 0 😆